Warning in flatpak-utils-http.c from clang analysis

From the "Build with clang and analyze" CI job:
```
common/flatpak-utils-http.c:259:7: warning: logical not is only applied to the left hand side of this comparison [-Wlogical-not-parentheses]
  if (!g_ascii_strncasecmp(buffer, header, hlen) == 0 ||
      ^                                          ~~
common/flatpak-utils-http.c:259:7: note: add parentheses after the '!' to evaluate the comparison first
  if (!g_ascii_strncasecmp(buffer, header, hlen) == 0 ||
      ^
       (                                             )
common/flatpak-utils-http.c:259:7: note: add parentheses around left hand side expression to silence this warning
  if (!g_ascii_strncasecmp(buffer, header, hlen) == 0 ||
      ^
      (                                         )
1 warning generated.
```

cc @alexlarsson

--- a/common/flatpak-utils-http.c
+++ b/common/flatpak-utils-http.c
@@ -257,7 +257,7 @@ check_header(char **value_out,
   if (realsize < hlen + 1)
     return;
 
-  if (!g_ascii_strncasecmp(buffer, header, hlen) == 0 ||
+  if (g_ascii_strncasecmp (buffer, header, hlen) != 0 ||
       buffer[hlen] != ':')
     return;
 

